### PR TITLE
Generate engine file during build

### PR DESCRIPTION
**Pull request by mocsharp**
Due to a bug that causes a distributed application to crash when the engine file is unavailable, we are adding an extra step in the CMAKE test step for the Distributed Endoscopy Tool Tracking application.  The new CMAKE test step is added as a required fixture for the Distribute Endoscopy Tool Tracking tests.

```bash
Test project /workspace/holohub/build/endoscopy_tool_tracking_distributed
    Start 1: endoscopy_engine_generation
1/5 Test #1: endoscopy_engine_generation ................................   Passed    0.23 sec
    Start 2: endoscopy_tool_tracking_distributed_cpp_test_args
2/5 Test #2: endoscopy_tool_tracking_distributed_cpp_test_args ..........   Passed    7.80 sec
    Start 3: endoscopy_tool_tracking_distributed_cpp_test_env_var
3/5 Test #3: endoscopy_tool_tracking_distributed_cpp_test_env_var .......   Passed    7.76 sec
    Start 4: endoscopy_tool_tracking_distributed_python_test_args
4/5 Test #4: endoscopy_tool_tracking_distributed_python_test_args .......   Passed    7.83 sec
    Start 5: endoscopy_tool_tracking_distributed_python_test_env_vars
5/5 Test #5: endoscopy_tool_tracking_distributed_python_test_env_vars ...   Passed    7.80 sec

100% tests passed, 0 tests failed out of 5
```

### CODE REVIEW

**Review by tbirdso — APPROVED** [review]

**Review by grlee77 — APPROVED** [review]
The approach looks good to me @mocsharp. Is this the only distributed app currently on Holohub? (just trying to determine if there is any other current app we should also do this for)

File: applications/endoscopy_tool_tracking_distributed/cpp/README.md [hunk 1]
```diff
@@ -21,13 +21,26 @@ If you want to manually convert the video data, please refer to the instructions
...
+# Build the Holohub container for the Distributed Endoscopy Tool Tracking application
+./dev_container build --docker_file applications/endoscopy_tool_tracking_distributed/Dockerfile
+
+# Launch the container
+./dev_container launch
+
+# Build the Distributed Endoscopy Tool Tracking application
+./run build endoscopy_tool_tracking_distributed
+
+# Generate the onnx engine
```
Comment (tbirdso) [inline applications/endoscopy_tool_tracking_distributed/cpp/README.md:35]: ONNX engine or TRT engine from ONNX file?

File: applications/endoscopy_tool_tracking_distributed/cpp/README.md [hunk 2]
```diff
@@ -21,13 +21,26 @@ If you want to manually convert the video data, please refer to the instructions
...
+python3 utilities/generate_trt_engine.py --input data/endoscopy/tool_loc_convlstm.onnx --output data/endoscopy/engines/ --fp16
+
 # Start the application with all three fragments
-./dev_container build_and_run endoscopy_tool_tracking_distributed --language cpp
+./run launch endoscopy_tool_tracking_distributed cpp
+
+# Once you have completed the above command, you may exit the container and
+#  use the following commands to run the application in distributed mode:
```
Comment (tbirdso) [inline applications/endoscopy_tool_tracking_distributed/cpp/README.md:42]: Is running with all three fragments required in advance? Or just TRT engine generation?

File: applications/endoscopy_tool_tracking_distributed/cpp/README.md [hunk 3]
```diff
@@ -21,13 +21,26 @@ If you want to manually convert the video data, please refer to the instructions
 
 ### Run Instructions
 
-Run the following command to start the application.  This will run the app with a pre-recorded video as input:
 
 ```sh
+# Build the Holohub container for the Distributed Endoscopy Tool Tracking application
+./dev_container build --docker_file applications/endoscopy_tool_tracking_distributed/Dockerfile
```
Comment (tbirdso) [inline applications/endoscopy_tool_tracking_distributed/cpp/README.md:27]: Would recommend building/running with non-default tag to minimize confusion among environments:

```suggestion
./dev_container build --docker_file applications/endoscopy_tool_tracking_distributed/Dockerfile --img holohub:endoscopy_tool_tracking_distributed
```

And then launch with `--img holohub:endoscopy_tool_tracking_distributed` below

**Review by tbirdso — COMMENTED** [review]
Fixture approach looks good. Verified tests pass locally. A couple of README suggestions.

**Review by tbirdso — APPROVED** [review]
Thanks @mocsharp !

**Pull request merged**